src/dogroups.c

```diff
@@ -143,6 +145,11 @@ SEXP dogroups(SEXP dt, SEXP dtcols, SEXP groups, SEXP grpcols, SEXP jiscols, SEX
         case REALSXP :
           REAL(VECTOR_ELT(SDall,j))[0] = NA_REAL;
           break;
+        case CPLXSXP :
```

Comment:
I don't know what this `switch` is doing, but here's how to initialize `CPLXSXP` -- set the `.r` and `.i` components to `NA_REAL` (taken from `r-source/src/main/complex.c`)